src/Mutation.tsx

```diff
@@ -0,0 +1,93 @@
+import * as React from 'react';
+import * as PropTypes from 'prop-types';
+import ApolloClient from 'apollo-client';
+const invariant = require('invariant');
+import { DocumentNode } from 'graphql';
+
+import { OperationVariables } from './types';
+
+// refetchQueries?: string[] | PureQueryOptions[];
+// update?: MutationUpdaterFn;
+// notifyOnNetworkStatusChange?: boolean;
+
+export interface MutationProps {
+  mutation: DocumentNode;
+  variables?: OperationVariables;
+  optimisticResponse?: Object;
+  children: (mutateFn: () => void, result?: any) => React.ReactNode;
```

Comment:
This result is the from `apollo-client` `mutate<T>(options: MutationOptions<T>): Promise<FetchResult<T>>`

Note there is a typing issue on it so it cannot yet be fully typed results: https://github.com/apollographql/apollo-client/issues/2795
  
Alternately, we could define our own `MutationResult` like we did in `Query` and solve the type problem ourselves.